packages/react-core/src/helpers/Popper/Popper.tsx

```diff
@@ -108,20 +116,33 @@ export const Popper: React.FunctionComponent<PopperProps> = ({
   onPopperClick,
   onDocumentKeyDown,
   enableFlip = true,
-  flipBehavior = 'flip'
+  flipBehavior = 'flip',
+  reference
 }) => {
   const [triggerElement, setTriggerElement] = React.useState(null);
+  const [refElement, setRefElement] = React.useState<HTMLElement>();
   const [popperElement, setPopperElement] = React.useState(null);
   const [ready, setReady] = React.useState(false);
-  const onDocumentClickCallback = React.useCallback(event => onDocumentClick(event, triggerElement, popperElement), [
-    isVisible,
-    triggerElement,
-    popperElement,
-    onDocumentClick
-  ]);
+  const onDocumentClickCallback = React.useCallback(
+    event => onDocumentClick(event, refElement || triggerElement, popperElement),
+    [isVisible, triggerElement, refElement, popperElement, onDocumentClick]
+  );
   React.useEffect(() => {
     setReady(true);
   }, []);
+  React.useEffect(() => {
+    if (reference) {
+      if ((reference as React.RefObject<any>).current) {
+        setRefElement((reference as React.RefObject<any>).current);
+      } else if (typeof reference === 'function') {
+        setRefElement(reference());
+      } else {
+        setRefElement(null);
+      }
+    } else {
+      setRefElement(null);
+    }
```

Comment:
Instead of these two branches, can't you set the default to null like `const [refElement, setRefElement] = React.useState<HTMLElement>(null);`?